dj/api/nodes.py

```diff
@@ -329,6 +284,75 @@ def create_node(
     return node  # type: ignore
 
 
+@router.post(
+    "/nodes/{name}/{column}/dimension/{dimension}/",
+    response_model=List[Column],
+)
+def add_dimension_to_node(
+    name: str,
+    column: str,
+    dimension: str,
```

Comment:
Nit: we could simplify this call a bit for columns that keep the same name as the dimension they reference, by letting users skip `dimension` argument. If None we could simply assume `dimension = column`.

Reply:
Hmm, I really like that idea but I'm wondering how to deal with this little tradeoff. Since it's a route parameter, it must be required. We could do something like:
```py
@app.post("/nodes/{name}/{column}/dimension/") 
@app.post("/nodes/{name}/{column}/dimension/{dimension}/")
async def root(name: str, column: str, dimension: Optional[str]=None):
...
```
Which is almost perfect, but for the first route, `dimension` will still show as an optional query parameter that can be used as `?dimension=foo` which can be confusing because now there's two ways to do this. I'm remembering now that you brought up in the sync that there will potentially be many things we'd want to do to a column, that feels like we should go full query parameter mode then and just have one endpoint for linking information to columns.
```py
@app.post("/nodes/{name}/{column}/")
async def root(name: str, column: str, dimension: Optional[str]=None, .....<more args here later?>....):
...
```
Then requests would look like `POST "/nodes/some_node/a_column_on_the_node?dimension=some_dimension_node"`. Thoughts?

Reply:
I prefer the latter, as that makes it clearer that it's not required and we'll always have that consistency with all our endpoints (i.e., if it's in the query parameter, it may or may not be required, but if it's in the route path, it's always required).

Reply:
Agree, the 2nd approach looks better and leaves more room for similar functionality later.

Reply:
Updated the PR to go with the second approach although added in "columns", otherwise the API gets confused with another route we have. That means `POST "/nodes/{name}/columns/{column}/"`